research.lua

```diff
@@ -0,0 +1,35 @@
+function on_research_tick(event)

```

Comment:
the player var is not accessible here, since it's the result of https://github.com/remijouannet/graftorio2/pull/24/files#diff-c7fb02f7fa6631c45d23ccf4b3c4b10dd012db736e9956129d9ebaa9aa18e756L4
i believe, you want to put on_research_tick in the event loop not after the loop

Reply:
does not look like this does the trick.

```lua
__graftorio2__/research.lua:31: attempt to index global 'player' (a nil value)
stack traceback:
	__graftorio2__/research.lua:31: in function 'on_research_tick'
	__graftorio2__/events.lua:80: in function <__graftorio2__/events.lua:1>
```

row 31:
```lua
for idx, tech in pairs(player.force.research_queue or {player.force.current_research}) do
```

(branch https://github.com/Kariton/graftorio2/tree/new_metrics-dev)

Reply:
row 24 does also utilize `'player'`

Reply:
I could remove it from the `player loop` and wrap it in its own...

Reply:
this is working.

Reply:
I got it working as intended with [2dc4abb](https://github.com/remijouannet/graftorio2/pull/24/commits/2dc4abb4b431afb06904bbd0c2c600d5f1339bb8)